sormas-ui/src/main/java/de/symeda/sormas/ui/caze/CaseDataForm.java

```diff
@@ -920,28 +909,41 @@ protected void addFields() {
 		});
 	}
 
-	private void onFollowUpUntilChanged(Property.ValueChangeEvent valueChangeEvent, DateField quarantineTo, CheckBox quarantineExtendedCheckBox) {
+	private void onFollowUpUntilChanged(Property.ValueChangeEvent valueChangeEvent, DateField quarantineTo, CheckBox quarantineExtendedCheckBox, CheckBox quarantineReducedCheckBox) {
 		Property<Date> followUpUntilField = valueChangeEvent.getProperty();
-		Date followUpUntil = followUpUntilField.getValue();
-		if (quarantineTo.getValue() != null && (followUpUntil == null || followUpUntil.compareTo(quarantineTo.getValue()) != 0)) {
+		Date newFollowUpUntil = followUpUntilField.getValue();
+		CaseDataDto originalCase = getInternalValue();
+		Date oldFollowUpUntil = originalCase.getFollowUpUntil();
+		Date oldQuarantineEnd = originalCase.getQuarantineTo();
+		if (adjustQuarantine(quarantineTo, newFollowUpUntil, oldFollowUpUntil)) {
 			VaadinUiUtil.showConfirmationPopup(
-				I18nProperties.getString(Strings.headingExtendQuarantine),
-				new Label(I18nProperties.getString(Strings.confirmationAlsoExtendQuarantine)),
+				I18nProperties.getString(Strings.headingAdjustQuarantine),
+				new Label(I18nProperties.getString(Strings.confirmationAlsoAdjustQuarantine)),
 				I18nProperties.getString(Strings.yes),
 				I18nProperties.getString(Strings.no),
 				640,
 				confirmed -> {
 					if (confirmed) {
 						quarantineChangedByFollowUpUntilChange = true;
-						quarantineTo.setValue(followUpUntil);
-						if (followUpUntil.after(getInternalValue().getFollowUpUntil())) {
-							quarantineExtendedCheckBox.setValue(true);
+						quarantineTo.setValue(newFollowUpUntil);
+						if (oldQuarantineEnd != null) {
+							boolean quarantineExtended = newFollowUpUntil.after(oldQuarantineEnd);
+							quarantineExtendedCheckBox.setValue(quarantineExtended);
+							quarantineReducedCheckBox.setValue(!quarantineExtended);
+							setVisible(quarantineExtended, CaseDataDto.QUARANTINE_EXTENDED);
+							setVisible(!quarantineExtended, CaseDataDto.QUARANTINE_REDUCED);
 						}
 					}
 				});
 		}
 	}
 
+	private boolean adjustQuarantine(DateField quarantineTo, Date newFollowUpUntil, Date oldFollowUpUntil) {
```

Comment:
The name of this method doesn't really make sense because it doesn't adjust anything, it just returns a boolean. Please find a name that is more appropriate for what this method does.